### ISSUE TITLE
Learning rate does not change despite setting decay variable

### ISSUE THREAD

**Issue by shahriar49** [keras-team/keras#13589]
**System information**  
- Have I written custom code (as opposed to using example directory):  Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Windows 10
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  1.12
- Keras version:  (bundled with TF 1.12)
- Python version:  3.6
- CUDA/cuDNN version:  N/A
- GPU model and memory:  N/A

**Describe the current behavior**  
I set both the learning rate and decay parameters for my SGD optimizer, for example:

`optimizer = tf.keras.optimizers.SGD(lr=0.1, decay=0.1/50)`

And then print learning rate before each epoch by defining a callback such as:
```
class PrintingCallback(tf.keras.callbacks.Callback):
    def on_epoch_begin(self, epoch, logs=None):
        print('Learning_rate is {}.'.format(K.get_value(self.model.optimizer.lr)))
``` 
But it always prints the initial learning rate at the start of each epoch without any change, while I think it should be decayed and the self.model.optimizer.lr variable should be updated.
